src/Sema/LowerZon.zig

```diff
@@ -144,7 +146,14 @@ fn lowerExprAnonResTy(self: *LowerZon, node: Zoir.Node.Index) CompileError!Inter
                     .any_aligned_fields = false,
                     .key = .{ .reified = .{
                         .zir_index = self.base_node_inst,
-                        .type_hash = @intFromEnum(node),
+                        .type_hash = hash: {
+                            var hasher: std.hash.Wyhash = .init(0);
+                            hasher.update(std.mem.asBytes(&node));
+                            hasher.update(std.mem.sliceAsBytes(elems));
+                            hasher.update(std.mem.sliceAsBytes(types));
```

Comment:
No need to do the types here, since they're determined entirely from the elements.